### PR TITLE
Update to work with PoET1

### PR DESCRIPTION
**Pull request by jjason**
Remove keyword arguments when calling into PoET enclave.
Renamed block_digest to block_hash everywhere to stay consistent with enclave.
Passing enclave wait timer object when creating wait certificate.
Fixed struct unpacking in PoET simulator that didn't work on Windows (but inexplicably worked on Linux).
Fix identifier property in wait certificate work across enclaves.
Force encoded PoET public key to string.
Fixed bug in wait timer duration creation.

Signed-off-by: Jamie Jason <jamie.jason@intel.com>

### CODE REVIEW

**Comment by cintel** [comment]
What is the motivation for moving to positional arguments?

**Comment by jjason** [comment]
Swig seems to be puking on it.

**Comment by jjason** [comment]
Want to get a fix in so @aludvik can proceed.  I personally like keyword arguments so will revisit to see if there is a better fix than simply going to positional arguments.

File: validator/sawtooth_validator/consensus/poet1/poet_enclave_simulator/poet_enclave_simulator.py [hunk 1]
```diff
@@ -343,7 +343,7 @@ def create_wait_timer(cls,
                         previous_certificate_id,
                         cls._seal_private_key))
 
-            tagd = float(struct.unpack('L', tag[-8:])[0]) / (2**64 - 1)
+            tagd = float(struct.unpack('Q', tag[-8:])[0]) / (2**64 - 1)
```
Comment (peterschwarz) [inline validator/sawtooth_validator/consensus/poet1/poet_enclave_simulator/poet_enclave_simulator.py]: Should be part of a separate commit, which explains the change from 'L' to 'Q' (this has nothing to do with keyword args).

File: validator/sawtooth_validator/consensus/poet1/wait_certificate.py [hunk 2]
```diff
@@ -120,10 +125,7 @@ def wait_certificate_from_serialized(cls,
 
     @property
     def enclave_wait_certificate(self):
-        return \
-            self.poet_enclave.deserialize_wait_certificate(
-                self._serialized_certificate,
-                self.signature)
+        return self._enclave_wait_certificate
```
Comment (peterschwarz) [inline validator/sawtooth_validator/consensus/poet1/wait_certificate.py]: Like wise, this change seems to be unrelated to the keyword args change.  Not clear on why the change is needed.

File: validator/sawtooth_validator/consensus/poet1/wait_timer.py [hunk 3]
```diff
@@ -150,6 +150,14 @@ def compute_local_mean(cls, certificates):
 
         return local_mean
 
+    @property
+    def enclave_wait_timer(self):
```
Comment (peterschwarz) [inline validator/sawtooth_validator/consensus/poet1/wait_timer.py]: Ditto

File: validator/tests/unit/poet1/test_wait_certificate.py [hunk 4]
```diff
@@ -78,7 +78,7 @@ def test_create_wait_certificate_before_wait_timer_expires(self):
                 certificates=[])
         wc = \
             WaitCertificate.create_wait_certificate(
-                wait_timer=None,
+                wait_timer=wt,
```
Comment (peterschwarz) [inline validator/tests/unit/poet1/test_wait_certificate.py]: Ditto

Comment (jjason) [inline validator/tests/unit/poet1/test_wait_certificate.py]: Will break out into separate commits.

**Comment by jjason** [comment]
@aludvik Was having problem with pickle and enclave wait certificate object.  Backed out a change that affected that.

If this works, will migrate any necessary changes to master.

File: validator/sawtooth_validator/consensus/poet1/wait_certificate.py [hunk 5]
```diff
@@ -75,11 +75,16 @@ def create_wait_certificate(cls,
...
-            cls.poet_enclave.create_wait_certificate(
-                wait_timer=wait_timer,
-                block_digest=block_digest)
+        enclave_certificate = None
+        try:
+            enclave_certificate = \
+                cls.poet_enclave.create_wait_certificate(
+                    wait_timer.enclave_wait_timer,
+                    block_hash)
+        except Exception as ex:
```
Comment (aludvik) [inline validator/sawtooth_validator/consensus/poet1/wait_certificate.py]: Should this swallow all exceptions here? Could cause a problem with debugging later. Is there a specific type of exception that you are expecting to see here for which you want enclave_certificate to be set to None?

Comment (jjason) [inline validator/sawtooth_validator/consensus/poet1/wait_certificate.py]: Yeah....it somehow got past the lint filters in the 0-7 branch, but got caught in master.  I need to "backport" tha fix.
